src/workerd/api/crypto/spkac.c++

```diff
@@ -1,102 +1,63 @@
-  BUF_MEM* bptr;
-  BIO_get_mem_ptr(bio, &bptr);
-  auto buf = jsg::BackingStore::alloc(js, bptr->length);
-  auto aptr = kj::arrayPtr(bptr->data, bptr->length);
-  buf.asArrayPtr<char>().copyFrom(aptr);
-  return jsg::BufferSource(js, kj::mv(buf));
-}
-
-kj::Maybe<kj::Own<NETSCAPE_SPKI>> tryGetSpki(kj::ArrayPtr<const kj::byte> input) {
-  static constexpr int32_t kMaxLength = kj::maxValue;
-  JSG_REQUIRE(input.size() <= kMaxLength, RangeError, "spkac is too large");
-  auto trimmed = trimTailingWhitespace(input.asChars());
-  auto ptr = NETSCAPE_SPKI_b64_decode(trimmed.begin(), trimmed.size());
-  if (!ptr) return kj::none;
-  return kj::disposeWith<NETSCAPE_SPKI_free>(ptr);
-}
-
-kj::Maybe<kj::Own<EVP_PKEY>> tryOwnPkey(kj::Own<NETSCAPE_SPKI>& spki) {
-  auto pkey = NETSCAPE_SPKI_get_pubkey(spki.get());
-  if (!pkey) return kj::none;
-  return kj::disposeWith<EVP_PKEY_free>(pkey);
-}
-
-kj::Maybe<kj::Own<BIO>> tryNewBio() {
-  auto bioptr = BIO_new(BIO_s_mem());
-  if (!bioptr) return kj::none;
-  return kj::disposeWith<BIO_free_all>(bioptr);
-}
-}  // namespace
 
 bool verifySpkac(kj::ArrayPtr<const kj::byte> input) {
   // So, this is fun. SPKAC uses MD5 as the digest algorithm. This is a problem because
   // using MD5 for signature verification is not allowed in FIPS mode, which means that
   // although we have a working implementation here, the result of this call is always
-  // going to false even if the input signature is correct. So this is a bit of a dead
-  // end that isn't going to be super useful. Fortunately but the exportPublicKey and
+  // going to be false even if the input signature is correct. So this is a bit of a dead
+  // end that isn't going to be super useful. Fortunately tho the exportPublicKey and
   // exportChallenge functions both work correctly and are useful. Unfortunately, this
   // likely means users would need to implement their own verification, which sucks.
   //
   // Alternatively we could choose to implement our own version of the validation that
   // bypasses BoringSSL's FIPS configuration. For now tho, this does end up matching
   // Node.js' behavior when FIPS is enabled so I guess that's something.
-  ClearErrorOnReturn clearErrorOnReturn;
   if (IoContext::hasCurrent()) {
     IoContext::current().logWarningOnce(
         "The verifySpkac function is currently of limited value in workers because "
         "the SPKAC signature verification uses MD5, which is not supported in FIPS mode. "
         "All workers run in FIPS mode. Accordingly, this method will currently always "
         "return false even if the SPKAC signature is valid. This is a known limitation.");
   }
-  KJ_IF_SOME(spki, tryGetSpki(input)) {
-    KJ_IF_SOME(key, tryOwnPkey(spki)) {
-      return NETSCAPE_SPKI_verify(spki.get(), key.get()) > 0;
-    }
-  }
-  return false;
+
+  ncrypto::Buffer<const char> buf{
+    .data = reinterpret_cast<const char*>(input.begin()),
+    .len = input.size(),
+  };
```

Comment:
Nit:

Can we just have a conversion function or a static method that converts to ncrypto::Buffer? It seems there are lots of repetitions.